components/tidb_query_expr/src/impl_time.rs

```diff
@@ -1600,10 +1601,74 @@ pub fn eval_from_unixtime(
     Ok(Some(tmp))
 }
 
+// unix_timestamp_to_mysql_unix_timestamp converts micto timestamp into MySQL's Unix timestamp.
+// MySQL's Unix timestamp ranges from '1970-01-01 00:00:01.000000' UTC to '3001-01-18 23:59:59.999999' UTC. Values out of range should be rewritten to 0.
+// https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_unix-timestamp
+fn unix_timestamp_to_mysql_unix_timestamp(micro_time: i64, frac: i8) -> Result<Decimal> {
+	if micro_time < 1000000 || micro_time > 32536771199999999 {
+		return Ok(Decimal::zero());
+	}
+
+    let time_in_decimal = Decimal::from_i64(micro_time)?;
+    let res = time_in_decimal.shift(-6);
+    let value = match res {
+        Res::Ok(value) => value,
+        Res::Truncated(_) => return Err(EvaluateError::Other("Decimal is truncated".to_string()).into()),
+        Res::Overflow(_) => return Err(EvaluateError::Other("Decimal is overflowed".to_string()).into()),
+    };
```

Comment:
impl<T> From<Res<T>> for Result<T> have been implemented.  We can simplify this part of the code, I think.